### PR TITLE
feat(checkbox): changes from review + added card variant #222

### PR DESCRIPTION
**Pull request by mart-sessman**
## Summary by CodeRabbit

* **New Features**
  * Added checkbox card component with primary and secondary variants
  * Introduced checkbox group component with horizontal and vertical layout support
  * Enhanced checkbox stories with new state matrices, card variants, icon support, and description handling

* **Documentation**
  * Added comprehensive contributor guides covering component creation, refactoring, testing, story authoring, and best practices
  * Updated accessibility checklist and communication guidelines

* **Tests**
  * Improved Jest testing environment to handle CSS parsing edge cases

* **Chores**
  * Added code review configuration and updated test setup

### Overview

The checkbox was rewritten from a self-contained wrapper component to a composable attribute directive on a native `<input>`. The consumer is now responsible for composing labels, feedback text, and layout.

This aligns with the TEDI-Ready design philosophy of composable primitives — native HTML elements styled via attribute directives, giving consumers full control over structure and accessibility.

**Import change:** `@tedi-design-system/angular/community` → `@tedi-design-system/angular/tedi`

### 1. CheckboxComponent — wrapper element → attribute directive (from prior release, unchanged here)

The core `CheckboxComponent` migration happened in a prior release. This does **not** change the TEDI-ready `CheckboxComponent` API. However, the community `CheckboxComponent` is now formally marked `@deprecated` in this release.

---

### 2. CheckboxGroupComponent — ControlValueAccessor wrapper → layout-only component

**What changed:** The community `tedi-checkbox-group` was a group component with `label`, `disabled`, `hasError`, `size`, `direction`, `spacing`, `feedbackText` inputs and `role="group"`. The TEDI-ready `tedi-checkbox-group` is a **layout-only** component with just `label` and `direction` inputs.

**Why:** Form control behavior is delegated to native checkbox inputs. The group only handles visual layout and labeling.

**Before (community):**
```html
<tedi-checkbox-group
  label="Pick options"
  direction="column"
  [spacing]="4"
  [disabled]="true"
  [hasError]="true"
  [feedbackText]="{ text: 'Required' }"
>
  <tedi-checkbox inputId="a" value="a">Option A</tedi-checkbox>
  <tedi-checkbox inputId="b" value="b">Option B</tedi-checkbox>
</tedi-checkbox-group>
```

**After (TEDI-ready):**
```html
<tedi-checkbox-group label="Pick options" direction="vertical">
  <label>
    <input type="checkbox" tedi-checkbox [invalid]="true" disabled [formControl]="controlA" />
    Option A
  </label>
  <label>
    <input type="checkbox" tedi-checkbox [invalid]="true" disabled [formControl]="controlB" />
    Option B
  </label>
  <tedi-feedback-text text="Required" />
</tedi-checkbox-group>
```

**Key changes:**
| Community input | TEDI-Ready equivalent | Notes |
|---|---|---|
| `label` | `label` | Kept |
| `direction` (`"row"` \| `"column"`) | `direction` (`"horizontal"` \| `"vertical"`) | Values renamed |
| `spacing` (number) | Removed | Handled by CSS tokens |
| `disabled` | Native `disabled` on each input | No group-level cascade |
| `hasError` | `[invalid]` on each input | No group-level cascade |
| `size` | `[size]` on each input | No group-level cascade |
| `feedbackText` input | `<tedi-feedback-text>` projected into group | Content projection slot |
| `role="group"` host attribute | Removed | Consumer adds if needed |

---

### 3. CheckboxCardGroupComponent — CVA compound → simple directive + layout wrapper

**What changed:** The community `tedi-checkbox-card-group` extended `CheckboxGroupComponent`, implemented `ControlValueAccessor` (managing a `string[]` value), and used `ChoiceGroupDirective` as a host directive to add `variant`, `hasIndicator`, and `spacing` inputs. The TEDI-ready approach splits this into two components: `tedi-checkbox-card` (attribute directive on `<label>`) and `tedi-checkbox-card-group` (pure layout wrapper with no inputs).

**Why:** Per-card control of variant and appearance is more flexible. Consumers manage their own form state.

**Before (community):**
```html
<tedi-checkbox-card-group
  label="Select plans"
  variant="secondary"
  [hasIndicator]="false"
  [spacing]="0"
  [formControl]="plansControl"
>
  <tedi-checkbox inputId="basic" value="basic">Basic</tedi-checkbox>
  <tedi-checkbox inputId="pro" value="pro">Pro</tedi-checkbox>
</tedi-checkbox-card-group>
```

**After (TEDI-ready):**
```html
<tedi-checkbox-card-group>
  <label tedi-checkbox-card variant="secondary" [showIndicator]="false">
    <input type="checkbox" tedi-checkbox [formControl]="basicControl" />
    Basic
  </label>
  <label tedi-checkbox-card variant="secondary" [showIndicator]="false">
    <input type="checkbox" tedi-checkbox [formControl]="proControl" />
    Pro
  </label>
</tedi-checkbox-card-group>
```

**Key changes:**
| Community | TEDI-Ready | Notes |
|---|---|---|
| `<tedi-checkbox-card-group>` with full API | `<tedi-checkbox-card-group>` (no inputs, pure layout) | |
| `variant` on group (via `ChoiceGroupDirective`) | `variant` on each `<label tedi-checkbox-card>` | Per-card |
| `hasIndicator` on group | `showIndicator` on each card | Renamed, default `true` |
| `spacing="0"` for grouped look | CSS handles gap | No `grouped` input (unlike radio-card) |
| `label` on group | Removed | Add your own label element |
| `ControlValueAccessor` (`string[]`) | None — bind `[formControl]` on individual inputs | Consumer manages array state |
| `value` on each checkbox | Removed | Consumer manages selection logic |
| `<tedi-checkbox>` children | `<input tedi-checkbox>` inside `<label tedi-checkbox-card>` | |

---

### 4. New Component: CheckboxCardComponent

This is a **new** component with no direct community predecessor (the community used `ChoiceGroupDirective` on the group, not on individual cards).

**Selector:** `label[tedi-checkbox-card]`

**Inputs:**
| Input | Type | Default |
|---|---|---|
| `variant` | `"primary"` \| `"secondary"` | `"primary"` |
| `showIndicator` | `boolean` | `true` |

**Usage:**
```html
<label tedi-checkbox-card variant="primary">
  <input type="checkbox" tedi-checkbox />
  Card label
  <tedi-feedback-text text="Description text" />
</label>
```

Content is projected into `.tedi-checkbox-card__content`, and `<tedi-feedback-text>` has a dedicated projection slot.

---

### 5. New Component: CheckboxGroupComponent (TEDI-ready)

**Selector:** `tedi-checkbox-group`

**Inputs:**
| Input | Type | Default |
|---|---|---|
| `label` | `string` | `undefined` |
| `direction` | `"horizontal"` \| `"vertical"` | `"horizontal"` |

**Usage:**
```html
<tedi-checkbox-group label="Options" direction="vertical">
  <label><input type="checkbox" tedi-checkbox /> Option 1</label>
  <label><input type="checkbox" tedi-checkbox /> Option 2</label>
  <tedi-feedback-text text="Select at least one" />
</tedi-checkbox-group>
```

---

### 6. Import Changes

**Before:**
```typescript
import {
  CheckboxComponent,
  CheckboxGroupComponent,
  CheckboxCardGroupComponent,
} from '@tedi-design-system/angular/community';
```

**After:**
```typescript
import {
  CheckboxComponent,
  CheckboxCardComponent,
  CheckboxGroupComponent,
  CheckboxCardGroupComponent,
} from '@tedi-design-system/angular/tedi';
```

---

### 7. Deprecations

The following community components are now marked `@deprecated`:
- `CheckboxComponent` (community) → use `CheckboxComponent` from `tedi`
- `CheckboxGroupComponent` (community) → use `CheckboxGroupComponent` from `tedi`
- `CheckboxCardGroupComponent` (community) → use `CheckboxCardComponent` + `CheckboxCardGroupComponent` from `tedi`

---

### Summary Table

| Community Component | TEDI-Ready Replacement | Selector Change |
|---|---|---|
| `<tedi-checkbox>` | `<input type="checkbox" tedi-checkbox>` | Element → attribute directive |
| `<tedi-checkbox-group>` (cascade API) | `<tedi-checkbox-group>` (layout only) | Same selector, reduced API |
| `<tedi-checkbox-card-group>` (CVA + ChoiceGroupDirective) | `<label tedi-checkbox-card>` + `<tedi-checkbox-card-group>` | Split into per-card directive + layout wrapper |

| Community Input | TEDI-Ready Equivalent |
|---|---|
| `inputId` | Native `id` |
| `value` (on checkbox) | Removed — manage in consumer |
| `checked` model | Native `checked` / `[formControl]` |
| `indeterminate` model | Native `[indeterminate]` |
| `disabled` model | Native `[disabled]` |
| `hasError` | `invalid` |
| `feedbackText` | `<tedi-feedback-text>` component |
| `size` (on group) | `size` (on each input) |
| `direction: "row"/"column"` | `direction: "horizontal"/"vertical"` |
| `spacing` | Removed (CSS tokens) |
| `variant` (on group) | `variant` (on each card) |
| `hasIndicator` (on group) | `showIndicator` (on each card, default `true`) |
| `ControlValueAccessor` (on card group) | Native `[formControl]` on individual inputs |

### CODE REVIEW

**Comment by mart-sessman** [comment]
@coderabbitai resolve

**Review by airikej — APPROVED** [review]

File: tedi/components/form/checkbox-group/checkbox-group.component.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+@if (label()) {
+  <p class="tedi-checkbox-group__label" tedi-text color="secondary">{{ label() }}</p>
+}
+<div class="tedi-checkbox-group__checks" [class.tedi-checkbox-group__checks--vertical]="direction() === 'vertical'">
+  <ng-content />
+</div>
+<div class="tedi-checkbox-group__subtexts">
+  <ng-content select="tedi-feedback-text" />
```
[the remark this answers is not in the document]
Comment (mart-sessman) [reply tedi/components/form/checkbox-group/checkbox-group.component.html:8]: not an issue
